future-architect/vuls
# add db backend redis

## What did you implement:

1. Update Package (go-cve-dictioanry and goval-dictionary)
1. Add db backend redis
1. change os family name string to const vars
1. use pointer for cve.Details (According to [go-cve-dictionary get method](https://github.com/kotakanbe/go-cve-dictionary/blob/0724974076170fa7d3f24252c1a26f71f8c763eb/db/db.go#L20)) 
1. Combine Database Validation method

## How did you implement it:

## How can we verify it:

- check redis backend

```
$ rm -rf vendor cache.db
```

```
$ dep ensure
```

```
$ vuls report -debug -to-localfile -format-full-text -cvedb-type=redis -cvedb-url="redis://localhost/0" -ovaldb-type=redis  -ovaldb-url="redis://localhost/1"
```

## Todos:
You don't have to satisfy all of the following.


***Is this ready for review?:*** YES 
***Is it a breaking change?:*** YES
